tests/ui/traits/dyn-iterator-deref-in-for-loop.rs

```diff
@@ -0,0 +1,15 @@
+//! Tests that dereferencing a Box<dyn Iterator> in a for loop correctly yields an error,
+//! as the unsized trait object does not implement IntoIterator.
+//! regression test for <https://github.com/rust-lang/rust/issues/20605>
+//@ revisions: current next
+//@ ignore-compare-mode-next-solver (explicit revisions)
+//@[next] compile-flags: -Znext-solver
+
+fn changer<'a>(mut things: Box<dyn Iterator<Item = &'a mut u8>>) {
+    for item in *things {
+        //~^ ERROR `dyn Iterator<Item = &'a mut u8>` is not an iterator
+        *item = 0
+    }
+}
```

Comment:
I have a quastion. 
Can it be considered the same as `for-loop-while/iter-from-mac-call.rs` and `traits/dyn-iterator-deref-in-for-loop`?
When this test was first added, there were no special flags. https://github.com/reddevilmidzy/rust/commit/3a62590be2bc479a32eb3285dc1024041ac02a5b